nifi-nar-bundles/nifi-amqp-bundle/nifi-amqp-processors/src/main/java/org/apache/nifi/amqp/processors/AMQPUtils.java

```diff
@@ -43,16 +41,48 @@
 
     private final static Logger logger = LoggerFactory.getLogger(AMQPUtils.class);
 
-    private final static List<String> propertyNames = Arrays.asList("amqp$contentType", "amqp$contentEncoding",
-            "amqp$headers", "amqp$deliveryMode", "amqp$priority", "amqp$correlationId", "amqp$replyTo",
-            "amqp$expiration", "amqp$messageId", "amqp$timestamp", "amqp$type", "amqp$userId", "amqp$appId",
-            "amqp$clusterId");
-    /**
-     * Returns a {@link List} of AMQP property names defined in
-     * {@link BasicProperties}
-     */
-    public static List<String> getAmqpPropertyNames() {
-        return propertyNames;
+    public enum PropertyNames {
```

Comment:
+1 here, primarily due to the fact that now individual properties (especially non-Sttring-value type) may need additional logic to help with conversion and therefore would need to be individually referenced in several places.